### PR TITLE
Add StaticPodOperatorSpec to operator/v1

### PR DESCRIPTION
**Pull request by damemi**
Adds a `failedRevisionLimit` and `succeededRevisionLimit` for static pod operators, based on the continuing work to implement these limits

### CODE REVIEW

**Comment by damemi** [comment]
Update from https://github.com/openshift/api/pull/128#issuecomment-453661222

This PR still needs generated updates but I'm having some trouble running them atm, opening the PR anyway to get eyes on it while I sort out my system

**Comment by damemi** [comment]
cc @deads2k

File: operator/v1/types.go [hunk 1]
```diff
@@ -175,6 +175,16 @@ const (
 	ConditionUnknown ConditionStatus = "Unknown"
 )
 
+// StaticPodOperatorSpec is spec for controllers that manage static pods.
+type StaticPodOperatorSpec struct {
+	OperatorSpec           `json:",inline"`
+
+	// failedRevisionLimit is the number of failed static pod installer revisions to keep on disk and in the api, with 0 being unlimited
+	FailedRevisionLimit    int `json:"failedRevisionLimit"`
```
Comment (deads2k) [inline operator/v1/types.go]: `FailedRevisionHistoryLimit`  similar down below.

We don't use ints, use int32

zero is not unlimited.  zero is the default.  -1 can be unlimited

omitempty

**Pull request merged**